docker-app/qfieldcloud/core/views/accounts_views.py

```diff
@@ -0,0 +1,50 @@
+from django.conf import settings
+from django.http import HttpRequest, HttpResponse, HttpResponseRedirect
+from django.contrib import messages
+from django.urls import reverse
+from django.utils.translation import gettext as _
+from allauth.account.models import EmailAddress
+from django.utils.http import url_has_allowed_host_and_scheme
+
+
+def resend_confirmation_email(request: HttpRequest) -> HttpResponse:
+    """
+    Resends a confirmation email to a new unverified user.
+    """
+    if request.method != "POST":
+        # referer is coming from the client, therefore it must be checked.
+        referer = request.headers.get("referer", "")
+        if url_has_allowed_host_and_scheme(
+            referer, allowed_hosts=settings.ALLOWED_HOSTS
+        ):
+            return HttpResponseRedirect(referer)
+        else:
+            return HttpResponseRedirect("account_login")
```

Comment:
What if we add a new function `redirect_to_referer_or_view(request: HttpRequest, view_name: str, *view_args, **view_kwargs)`?